librad/src/git/types.rs

```diff
@@ -114,6 +115,14 @@ impl<R> SymbolicRef<R> {
     }
 }
 
+/// A `Display`-like trait but only meant for `Refspec`s.
```

Comment:
I want to be able to turn `Refspec`s into `String`s dynamically but not allow anything that implements `ToString` or `Display`.

I'll try be more verbose in this doc :)

Reply:
Something about this trips me up -- there's a lot of generic machinery to make these things typed, but then we're actually unifying with `String`. Is there not some intermediate representation of a refspec which includes all the permutations on the value level (like, `Option` fields), so we can have _one_ `Display` impl? The job of the polymorphic types would then be to make construction of this _safe_ depending on the context.

Reply:
Well, there's only one `Display` impl. Whereas this is a bit of a hack to get collections of `Refspec`s without the types clashing.

> Is there not some intermediate representation of a refspec which includes all the permutations on the value level (like, Option fields)

Can you expand on what you mean a bit? I'm not quite following.

But I agree, it's a bit awkward because we don't have something like an open union to extend out the cases.